presto-raptor/src/main/java/com/facebook/presto/raptor/systemtables/ShardMetadataRecordCursor.java

```diff
@@ -328,6 +330,26 @@ else if (temporalType.equals(TIMESTAMP)) {
         return builder.build();
     }
 
+    private static void buildPredicatesAndValues(Domain nameDomain, List<String> values, List<String> predicates, String predicatePrefix)
+    {
+        if (nameDomain.isSingleValue()) {
+            predicates.add(predicatePrefix + " = ?");
+            values.add(getStringValue(nameDomain.getSingleValue()));
+        }
+        else {
+            List<String> names = nameDomain.getValues().getRanges().getOrderedRanges().stream()
+                    .filter(Range::isSingleValue)
+                    .map(range-> getStringValue(range.getSingleValue()))
+                    .collect(toList());
+            // Add a predicate only when nameDomain has actual content. Cases like "select * from system.shards where table_name <'ab';" will be ingnored.
+            if (!names.isEmpty()) {
+                values.addAll(names);
+                String inClause = Joiner.on(",").join(Collections.nCopies(names.size(), "?"));
+                predicates.add(String.format("%s IN ( %s )", predicatePrefix, inClause));
```

Comment:
Nit: we want the output SQL to be formatted like `xxx IN (?, ?)`, so remove the spaces around the parens

Also, static import `format`